### DISCUSSION TITLE
Can't access Page Component props (e.g. getLayout) when using a HOC

### DISCUSSION THREAD

**Discussion by anicioalexandre** [vercel/next.js#33468] [Help] [answered]
Hello guys, on my app I have several pages that uses some HOCs to access context as props, like this one:

**`ComponentPage.jsx:`**
```
import withSomeContext from 'utils/withSomeContext'

const ComponentPage = ({states, actions) => {
  return (...)
}

ComponentPage.getLayout = function getLayout(page) {
  return (
    <div>
      Some Header
      <div>{page}</div>
    </div>
  )
}

export default withSomeContext(ComponentPage)
```
But then when I pass some props like `getLayout` I can't really retrieve it on the `_app` because of that withSomeContext HOC:

**`_app.js:`**
```
function MyApp({Component, pageProps}) {
  console.log(Component.getLayout, 'I can't retrieve my getLayout prop') ---> undefined

  return (
    <...>
    (...)
    <AppContainer>
        <Component {...pageProps} />
    </AppContainer>
    </...>
  )
}

export default MyApp
```
This happens because the `Component` prop on the `_app` sees is my `withSomeContext` HOC function instead of my real Page Component. Do you guys can think on any workaround for being able to actually retrieve these kind of props we set for a Page Component while using a HOC to render it? Am I missing something?

**`withSomeContext.js:`**
```
import {SomeContextConsumer} from 'contexts/SomeContextConsumer'

export const withProperties = (Component) => (props) => {
  return (
    <SomeContextConsumer>
      <Component {...props} />
    </SomeContextConsumer>
  )
}
```

**Comment by guilhermeoac** [comment, marked as answer]
I'm having the exact same problem

**Reply by HamzaAmar** [reply]
hello @guilhermeoac  @anicioalexandre  You can fix this problem assign get Layout Not ComponentPage But To the new HOC Components because it the direct child of app 

```
  import withSomeContext from 'utils/withSomeContext'
  
  const ComponentPage = ({states, actions) => {
    return (...)
  }
  
  const CompWithContext = withSomeContext(ComponentPage)
  
  export default CompWithContext 
  
  CompWithContext.getLayout = function getLayout(page) {
    return (
      <div>
        Some Header
        <div>{page}</div>
      </div>
    )
  }
```

**Reply by anicioalexandre** [reply]
Nice one @HamzaAmar, thank you!!

**Reply by jfilipeferreira96** [reply]
I did not get the answer, is there any way you could simplify it or explain it better with your case @anicioalexandre ? Thanks!

**Reply by jfilipeferreira96** [reply]
Oh nevermind, I figured it out. Thanks @HamzaAmar!
